### PR TITLE
Token obtaining within CourseService

### LINKED ISSUE #344: Fix accessToken obtaining in CourseService in AdminPanel
Add IDataProtector to CourseService constructor, setup accessToken obtainin, remove accessToken from Service methods parameter.

### PR DESCRIPTION
**Pull request by andy-dendy-repo**
Change CourseService interface method, which takes an accesse token as parameter. 
Create Token obtaining logic inside of CourseService similar to other services in AdminPanel.

### CODE REVIEW

**Review by QuestroL — APPROVED** [review]

**Review by Serhii-Dudar — APPROVED** [review]

File: CharlieBackend.AdminPanel/Services/CourseService.cs [hunk 1]
```diff
@@ -19,16 +21,22 @@ public class CourseService: ICourseService
...
+
+        public CourseService(IApiUtil apiUtil, IOptions<ApplicationSettings> config, IMapper mapper, 
+            IHttpContextAccessor httpContextAccessor, IDataProtectionProvider provider)
         {
             _apiUtil = apiUtil;
             _config = config;
             _mapper = mapper;
+
+            IDataProtector protector = provider.CreateProtector(_config.Value.Cookies.SecureKey);
+            _accessToken = protector.Unprotect(httpContextAccessor.HttpContext.Request.Cookies["accessToken"]);
```
Comment (gingerbread-monster) [inline CharlieBackend.AdminPanel/Services/CourseService.cs:34]: What if it fails to decypher access token obtained from inside the cookie? (And why is it encrypted in the first place? What is the point of encrypting JWT?)

If `Unprotect()` method throws an [exception](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.dataprotection.idataprotector.unprotect?view=aspnetcore-5.0#exceptions), then exception handler here
https://github.com/ita-social-projects/WhatBackend/blob/0f78082fd6c8f03ede68ab971f5a5d0371028fc6/CharlieBackend.AdminPanel/Middlewares/ExceptionHandleMiddleware.cs#L33
will end up writing *HTTP 500: Internal Server Error* response to a client.
Correct me if I'm wrong.

**Comment by gingerbread-monster** [comment]
> 
> 
> _No description provided._

Please, always provide a short description of your changes inside your pull request.

File: CharlieBackend.AdminPanel/Services/CourseService.cs (hunk 1 above)
Comment (andy-dendy-repo) [reply CharlieBackend.AdminPanel/Services/CourseService.cs:34]: Yes, an undescribed error would occure. But it is a common way to obtain tokey through all the Services in AdminPanel.
I've suggested this changes to continue work with Views. Which needs a solid set of Service interfaces.
I think, the inner realisation of All Services should be enhaced in futher development at one pull request.

**Review by Hlib-mats — APPROVED** [review]

**Comment by andy-dendy-repo** [comment]
> > _No description provided._
> 
> Please, always provide a short description of your changes inside your pull request.

Solved, thnks.

**Review by QuestroL — APPROVED** [review]

**Review by Hlib-mats — APPROVED** [review]

**Review by gingerbread-monster — APPROVED** [review]

**Review by CptHouston — APPROVED** [review]

**Pull request merged**